tools/container/toolkit/toolkit.go

```diff
@@ -436,6 +451,24 @@ func installToolkitConfig(c *cli.Context, toolkitConfigPath string, nvidiaContai
 	// Use the driver run root as the root:
 	driverLdconfigPath := config.NormalizeLDConfigPath("@" + filepath.Join(opts.DriverRoot, strings.TrimPrefix(ldconfigPath, "@/")))
 
+	var ctrRuntimeBinPaths []string
```

Comment:
As I already mentioned, we should pass these in. Looking at the implementation, we ALREADY have an option to pass this information. 

What about something like:
```
diff --git a/pkg/config/engine/api.go b/pkg/config/engine/api.go
index 449ceb48..aa6fa853 100644
--- a/pkg/config/engine/api.go
+++ b/pkg/config/engine/api.go
@@ -30,3 +30,25 @@ type Interface interface {
 type RuntimeConfig interface {
 	GetBinPath() string
 }
+
+// TODO: We could consider adding this to the Interface type.
+func GetBinaryPathsForRuntimes(cfg Interface, runtimes ...string) []string {
+	var binaryPaths []string
+
+	// We use a map to ensure that we don't add duplicate paths, but maintain ordering.
+	seen := make(map[string]bool)
+	for _, runtime := range runtimes {
+		runtimeConfig, err := cfg.GetRuntimeConfig(runtime)
+		if err != nil {
+			continue
+		}
+		binaryPath := runtimeConfig.GetBinPath()
+		if binaryPath == "" || seen[binaryPath] {
+			continue
+		}
+		seen[binaryPath] = true
+		binaryPaths = append(binaryPaths, binaryPath)
+	}
+
+	return binaryPaths
+}
diff --git a/tools/container/nvidia-toolkit/run.go b/tools/container/nvidia-toolkit/run.go
index f6dfe08c..7b7edf4f 100644
--- a/tools/container/nvidia-toolkit/run.go
+++ b/tools/container/nvidia-toolkit/run.go
@@ -164,7 +164,13 @@ func Run(c *cli.Context, o *options) error {
 	}
 	defer shutdown(o.pidFile)
 
-	err = toolkit.Install(c, &o.toolkitOptions, o.toolkitRoot(), o.runtime, o.runtimeOptions.HostRootMount)
+	lowlevelRuntimePaths, err := runtime.GetLowlevelRuntimePaths(&o.runtimeOptions, o.runtime)
+	if err != nil {
+		return fmt.Errorf("unable to determine runtime options: %w", err)
+	}
+
+	o.toolkitOptions.ContainerRuntimeRuntimes = *cli.NewStringSlice(lowlevelRuntimePaths...)
+	err = toolkit.Install(c, &o.toolkitOptions, o.toolkitRoot())
 	if err != nil {
 		return fmt.Errorf("unable to install toolkit: %v", err)
 	}
diff --git a/tools/container/runtime/containerd/containerd.go b/tools/container/runtime/containerd/containerd.go
index cb154ac7..98234517 100644
--- a/tools/container/runtime/containerd/containerd.go
+++ b/tools/container/runtime/containerd/containerd.go
@@ -23,6 +23,7 @@ import (
 	log "github.com/sirupsen/logrus"
 	cli "github.com/urfave/cli/v2"
 
+	"github.com/NVIDIA/nvidia-container-toolkit/pkg/config/engine"
 	"github.com/NVIDIA/nvidia-container-toolkit/pkg/config/engine/containerd"
 	"github.com/NVIDIA/nvidia-container-toolkit/tools/container"
 )
@@ -166,3 +167,20 @@ func (o *Options) runtimeConfigOverride() (map[string]interface{}, error) {
 
 	return runtimeOptions, nil
 }
+
+func GetLowlevelRuntimePaths(hostRoot string) ([]string, error) {
+	cfg, err := containerd.New(
+		containerd.WithConfigSource(containerd.CommandLineSource(hostRoot)),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to load containerd config: %w", err)
+	}
+
+	var runtimes []string
+	if defaultRuntime := cfg.DefaultRuntime(); defaultRuntime != "" {
+		runtimes = append(runtimes, defaultRuntime)
+	}
+	runtimes = append(runtimes, "runc")
+
+	return engine.GetBinaryPathsForRuntimes(cfg, runtimes...), nil
+}
diff --git a/tools/container/runtime/runtime.go b/tools/container/runtime/runtime.go
index 6fa5442a..44ea2577 100644
--- a/tools/container/runtime/runtime.go
+++ b/tools/container/runtime/runtime.go
@@ -166,3 +166,12 @@ func Cleanup(c *cli.Context, opts *Options, runtime string) error {
 		return fmt.Errorf("undefined runtime %v", runtime)
 	}
 }
+
+func GetLowlevelRuntimePaths(opts *Options, runtime string) ([]string, error) {
+	switch runtime {
+	case containerd.Name:
+		return containerd.GetLowlevelRuntimePaths(opts.HostRootMount)
+	default:
+		return nil, fmt.Errorf("undefined runtime %v", runtime)
+	}
+}
diff --git a/tools/container/toolkit/toolkit.go b/tools/container/toolkit/toolkit.go
index 9782df9d..d05405af 100644
--- a/tools/container/toolkit/toolkit.go
+++ b/tools/container/toolkit/toolkit.go
@@ -31,9 +31,6 @@ import (
 
 	"github.com/NVIDIA/nvidia-container-toolkit/internal/config"
 	"github.com/NVIDIA/nvidia-container-toolkit/internal/system/nvdevices"
-	"github.com/NVIDIA/nvidia-container-toolkit/pkg/config/engine"
-	"github.com/NVIDIA/nvidia-container-toolkit/pkg/config/engine/containerd"
-	"github.com/NVIDIA/nvidia-container-toolkit/pkg/config/engine/crio"
 	"github.com/NVIDIA/nvidia-container-toolkit/pkg/nvcdi"
 	transformroot "github.com/NVIDIA/nvidia-container-toolkit/pkg/nvcdi/transform/root"
 )
@@ -290,7 +287,7 @@ func TryDelete(cli *cli.Context, toolkitRoot string) error {
 
 // Install installs the components of the NVIDIA container toolkit.
 // Any existing installation is removed.
-func Install(cli *cli.Context, opts *Options, toolkitRoot, runtime, hostRoot string) error {
+func Install(cli *cli.Context, opts *Options, toolkitRoot string) error {
 	log.Infof("Installing NVIDIA container toolkit to '%v'", toolkitRoot)
 
 	log.Infof("Removing existing NVIDIA container toolkit installation")
@@ -354,7 +351,7 @@ func Install(cli *cli.Context, opts *Options, toolkitRoot, runtime, hostRoot str
 	}
 
 	err = installToolkitConfig(cli, toolkitConfigPath, nvidiaContainerCliExecutable, nvidiaCTKPath,
-		nvidiaContainerRuntimeHookPath, runtime, hostRoot, opts)
+		nvidiaContainerRuntimeHookPath, opts)
 	if err != nil && !opts.ignoreErrors {
 		return fmt.Errorf("error installing NVIDIA container toolkit config: %v", err)
 	} else if err != nil {
@@ -429,7 +426,7 @@ func installLibrary(libName string, toolkitRoot string) error {
 // installToolkitConfig installs the config file for the NVIDIA container toolkit ensuring
 // that the settings are updated to match the desired install and nvidia driver directories.
 func installToolkitConfig(c *cli.Context, toolkitConfigPath string, nvidiaContainerCliExecutablePath string, nvidiaCTKPath string,
-	nvidaContainerRuntimeHookPath string, runtime, hostRoot string, opts *Options) error {
+	nvidaContainerRuntimeHookPath string, opts *Options) error {
 	log.Infof("Installing NVIDIA container toolkit config '%v'", toolkitConfigPath)
 
 	cfg, err := config.New(
@@ -451,24 +448,6 @@ func installToolkitConfig(c *cli.Context, toolkitConfigPath string, nvidiaContai
 	// Use the driver run root as the root:
 	driverLdconfigPath := config.NormalizeLDConfigPath("@" + filepath.Join(opts.DriverRoot, strings.TrimPrefix(ldconfigPath, "@/")))
 
-	var ctrRuntimeBinPaths []string
-	ctrRuntimeBinPaths, err = getRuntimeBinaryPaths(runtime, hostRoot)
-	if err != nil {
-		log.Warningf("Error retrieving runtime binary paths: %v", err)
-	}
-
-	var ctkRuntimes []string
-	defaultCfg, err := cfg.Config()
-	if err == nil {
-		defaultCfgRuntimes := defaultCfg.NVIDIAContainerRuntimeConfig.Runtimes
-		if len(ctrRuntimeBinPaths) > 0 {
-			ctkRuntimes = append(ctkRuntimes, ctrRuntimeBinPaths...)
-		}
-		ctkRuntimes = append(ctkRuntimes, defaultCfgRuntimes...)
-	} else {
-		log.Warningf("could not get default toolkit config: %v", err)
-	}
-
 	configValues := map[string]interface{}{
 		// Set the options in the root toml table
 		"accept-nvidia-visible-devices-envvar-when-unprivileged": opts.acceptNVIDIAVisibleDevicesWhenUnprivileged,
@@ -484,10 +463,6 @@ func installToolkitConfig(c *cli.Context, toolkitConfigPath string, nvidiaContai
 		"nvidia-container-runtime-hook.skip-mode-detection": opts.ContainerRuntimeHookSkipModeDetection,
 	}
 
-	if len(ctkRuntimes) > 0 {
-		configValues["nvidia-container-runtime.runtimes"] = ctkRuntimes
-	}
-
 	for key, value := range configValues {
 		cfg.Set(key, value)
 	}
@@ -821,87 +796,3 @@ func generateCDISpec(opts *Options, nvidiaCDIHookPath string) error {
 
 	return nil
 }
-
-// getRuntimeBinaryPaths extracts the full paths of the low-level runtime binaries specified in the container runtime config
-func getRuntimeBinaryPaths(runtime, hostRoot string) ([]string, error) {
-	var runtimeBinaryPaths []string
-
-	var err error
-
-	switch runtime {
-	case "containerd":
-		runtimeBinaryPaths, err = getContainerdRuntimeBinaryPaths(hostRoot)
-		if err != nil {
-			log.Warningf("Unable to determine containerd runtime binary paths: %v", err)
-		}
-
-	case "crio":
-		runtimeBinaryPaths, err = getCrioRuntimeBinaryPaths(hostRoot)
-		if err != nil {
-			log.Warningf("Unable to determine crio runtime binary paths: %v", err)
-		}
-	}
-
-	return runtimeBinaryPaths, nil
-}
-
-// getContainerdRuntimeBinaryPaths extracts the full paths of the low-level runtime binaries specified in the containerd runtime config
-func getContainerdRuntimeBinaryPaths(hostRoot string) ([]string, error) {
-	cfg, err := containerd.New(
-		containerd.WithConfigSource(containerd.CommandLineSource(hostRoot)),
-	)
-	if err != nil {
-		return nil, fmt.Errorf("unable to load containerd config: %w", err)
-	}
-
-	runtimeHandlers := []string{"runc"}
-	defaultRuntime := cfg.DefaultRuntime()
-	if defaultRuntime != "" {
-		runtimeHandlers = append(runtimeHandlers, defaultRuntime)
-	}
-
-	return getBinaryPathsForHandlers(cfg, runtimeHandlers), nil
-}
-
-// getCrioRuntimeBinaryPaths extracts the full paths of the low-level runtime binaries specified in the CRI-O runtime config
-func getCrioRuntimeBinaryPaths(hostRoot string) ([]string, error) {
-	cfg, err := crio.New(
-		crio.WithConfigSource(crio.CommandLineSource(hostRoot)),
-	)
-	if err != nil {
-		return nil, fmt.Errorf("unable to load crio config: %w", err)
-	}
-
-	var runtimeHandlers []string
-	defaultRuntime := cfg.DefaultRuntime()
-	if defaultRuntime != "" {
-		runtimeHandlers = append(runtimeHandlers, defaultRuntime)
-	}
-	runtimeHandlers = append(runtimeHandlers, "runc")
-
-	return getBinaryPathsForHandlers(cfg, runtimeHandlers), nil
-}
-
-func getBinaryPathsForHandlers(cfg engine.Interface, handlers []string) []string {
-	var binaryPathsSet []string
-
-	// We use a map to create a string set (unique string list) of binary paths
-	binaryPathsMap := map[string]struct{}{}
-	for _, rh := range handlers {
-		runtimeCfg, err := cfg.GetRuntimeConfig(rh)
-		if err != nil {
-			log.Warningf("Unable to determine runtime binary path for %s: %v", rh, err)
-			continue
-		}
-		binPath := runtimeCfg.GetBinPath()
-		if binPath != "" {
-			binaryPathsMap[binPath] = struct{}{}
-		}
-	}
-
-	for b := range binaryPathsMap {
-		binaryPathsSet = append(binaryPathsSet, b)
-	}
-
-	return binaryPathsSet
-}
```